### DISCUSSION TITLE
v2 planning

### DISCUSSION THREAD

**Discussion by earthboundkid** [earthboundkid/requests#110] [Ideas]
The package URL for this repo is now wonky between carlmjohnson and earthboundkid due to the user rename. I was thinking there wasn't a good way to fix this, but if the package moves to v2, the module path can be changed without annoying old users who try to upgrade their dependencies and get a module name mismatch error message. 

If we move to v2 this, we might as well fix other problems like #81 at the same time. 

What else should we clean up as part of v2?

**Comment by earthboundkid** [comment]
Drop ToHeaders and just have CopyHeaders.

**Comment by earthboundkid** [comment]
Make BodyJSON etc into a Config.

**Comment by earthboundkid** [comment]
Remove minitrue and slicex as part of going to Go 1.23 or whatever is out when this happens.

**Comment by earthboundkid** [comment]
I should make an organization, so this can be a one time transfer.

**Comment by earthboundkid** [comment]
Do #123 and drop the old functions.

**Comment by PhakornKiong** [comment]
Would love to see a more explicit way of extracting body of requests when it is considered error.

Related to 
https://github.com/earthboundkid/requests/discussions/40

**Reply by earthboundkid** [reply]
This was added with [ValidationHandler](https://pkg.go.dev/github.com/carlmjohnson/requests#example-ValidatorHandler):

```go
// If we fail validation because the response is a 404,
// we handle the body with errBody instead of regularBody
// for separate processing.
err := requests.
	URL("http://example.com/404").
	ToString(&regularBody).
	AddValidator(
		requests.ValidatorHandler(
			requests.DefaultValidator,
			requests.ToString(&errBody),
		)).
	Fetch(context.Background())
```
